src/layout/DwindleLayout.cpp

```diff
@@ -108,11 +108,18 @@ void CHyprDwindleLayout::applyNodeDataToWindow(SDwindleNodeData* pNode, bool for
     const bool DISPLAYTOP    = STICKS(pNode->position.y, PMONITOR->vecPosition.y + PMONITOR->vecReservedTopLeft.y);
     const bool DISPLAYBOTTOM = STICKS(pNode->position.y + pNode->size.y, PMONITOR->vecPosition.y + PMONITOR->vecSize.y - PMONITOR->vecReservedBottomRight.y);
 
-    const auto PBORDERSIZE = &g_pConfigManager->getConfigValuePtr("general:border_size")->intValue;
-    const auto PGAPSIN     = &g_pConfigManager->getConfigValuePtr("general:gaps_in")->intValue;
-    const auto PGAPSOUT    = &g_pConfigManager->getConfigValuePtr("general:gaps_out")->intValue;
-
     const auto PWINDOW = pNode->pWindow;
+    // get specific gaps and rules for this workspace,
+    // if user specified them in config
+    const auto         WORKSPACE_RULE = g_pConfigManager->getWorkspaceRuleFor(g_pCompositor->getWorkspaceByID(PWINDOW->m_iWorkspaceID));
+
+    static auto* const PGAPSIN     = &g_pConfigManager->getConfigValuePtr("general:gaps_in")->intValue;
+    static auto* const PGAPSOUT    = &g_pConfigManager->getConfigValuePtr("general:gaps_out")->intValue;
+    static auto* const PBORDERSIZE = &g_pConfigManager->getConfigValuePtr("general:border_size")->intValue;
+
+    auto               gapsIn     = WORKSPACE_RULE.gapsIn.value_or(*PGAPSIN);
+    auto               gapsOut    = WORKSPACE_RULE.gapsOut.value_or(*PGAPSOUT);
+    auto               borderSize = WORKSPACE_RULE.borderSize.value_or(*PBORDERSIZE);
```

Comment:
wait, not clear to me. The bordersize behaviour should be unaffacted by my code right? was this a pre-existing issue?

Anyways, I need yet again some guidance here. I can simply add a field to the `SWindowSpecialRenderData` struct, but where should this be used then? I assume something has to change in the `CWindow::updateWindowDecos()` somewhere, but wouldn't really know how or what.

Reply:
windows never had bordersize props. The border size here is only used to calculate gaps, because gaps = gaps_in/out + borderSize

> but where should this be used then

CHyprRenderer::renderWindow